src/Lykke.Job.CandlesHistoryWriter.Services/HistoryMigration/MidPriceFixManager.cs

```diff
@@ -0,0 +1,255 @@
+
+                //replace candles from Sec to bigger intervals
+                foreach (var candleBatch in candlesByInterval)
+                {
+                    var candlesToReplace = candleBatch.Value.ToList();
+                    await _candlesHistoryRepository.ReplaceCandlesAsync(candlesToReplace);
+                    Health.FixedCandlesCount += candlesToReplace.Count;
+                }
+
+                _log.Info(nameof(FixMidPricesAsync),
+                    $"Mid candle prices fix for {assetPairId} finished.");
+            }
+            catch (Exception ex)
+            {
+                Health.Errors.Add($"{assetPairId}: {ex.Message}");
+                _log.Error(nameof(FixMidPricesAsync), ex);
+            }
+        }
+
+        private async Task<IReadOnlyList<ICandle>> TryGetCorruptedCandlesAsync(string assetPairId, int assetPairAccuracy)
+        {
+            var (dateFrom, dateTo) = await _candlesFiltrationService.GetDateTimeRangeAsync(assetPairId, CandlePriceType.Mid);
+
+            if (dateFrom == DateTime.MinValue || dateTo == DateTime.MinValue)
+                return new List<ICandle>();
+
+            var currentMonthBeginingDateTime = dateTo;
+
+            var result = new List<ICandle>();
+
+            var (bidMonthCandles, askMonthCandles, midMonthCandles) = await GetCandlesAsync(assetPairId, CandleTimeInterval.Month, dateFrom, dateTo);
+
+            IReadOnlyList<ICandle> lastCandles = GetCorruptedCandlesWithFixedData(bidMonthCandles, askMonthCandles, midMonthCandles, assetPairAccuracy);
+
+            // There are no incorrect candles at all - returning.
+            if (!lastCandles.Any())
+                return result;
+
+            result.AddRange(lastCandles);
+
+            Health.CorruptedCandlesCount += lastCandles.Count;
+
+            for (var i = Constants.DbStoredIntervals.Length - 2; i >= 0; i--)
+            {
+                var currentCandles = new List<ICandle>();
+                var interval = Constants.DbStoredIntervals[i];
+
+                Health.CurrentInterval = interval;
+
+                foreach (var candle in lastCandles)
+                {
+                    dateFrom = candle.Timestamp.TruncateTo(interval); // Truncating is important when searching weeks by months: the first week of month may start earlier than the month. In other cases, TruncateTo is redundant.
+                    dateTo = candle.Timestamp.AddIntervalTicks(1, GetBiggerInterval(interval));
+                    if (dateTo > currentMonthBeginingDateTime)
+                        dateTo = currentMonthBeginingDateTime;
+
+                    Health.DateFrom = dateFrom;
+                    Health.DateTo = dateTo;
+
+                    var (bidCandles, askCandles, midCandles) = await GetCandlesAsync(assetPairId, interval, dateFrom, dateTo);
+
+                    var corruptedCandles = GetCorruptedCandlesWithFixedData(bidCandles, askCandles, midCandles, assetPairAccuracy);
```

Comment:
Is it corrupted of fixed candles? names of the method and variable are inconsistent (and containing method too, and variables to which its result is assigned to)